src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -94,7 +143,156 @@ void DlgPrefColors::slotApply() {
                         m_colorPaletteSettings.getTrackColorPalette()));
     }
 
-    m_pConfig->setValue(
-            ConfigKey("[Controls]", "auto_hotcue_colors"),
-            checkBoxAssignHotcueColors->isChecked());
+    int index = comboBoxHotcueDefaultColor->currentIndex();
+
+    if (index > 0) {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), index - 1);
+    } else {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), true);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), -1);
+    }
+}
+
+QPixmap DlgPrefColors::drawPalettePreview(const QString& paletteName) {
+    ColorPalette palette = m_colorPaletteSettings.getHotcueColorPalette(paletteName);
+    if (paletteName == palette.getName()) {
+        QPixmap pixmap(kPalettePreviewSize);
+        int count = math_max(palette.size(), 1);
+        // Rounding up is required so the entire width of the pixmap is filled up to the edge.
+        int widthPerColor = ceil(pixmap.width() / static_cast<float>(count));
+        QPainter painter(&pixmap);
+        for (int i = 0; i < palette.size(); ++i) {
+            painter.setPen(mixxx::RgbColor::toQColor(palette.at(i)));
+            painter.setBrush(mixxx::RgbColor::toQColor(palette.at(i)));
+            painter.drawRect(i * widthPerColor, 0, widthPerColor, pixmap.height());
+        }
+        return pixmap;
+    }
+    return QPixmap();
+}
+
+QIcon DlgPrefColors::drawPaletteIcon(const QString& paletteName) {
+    QPixmap pixmap(16, 16);
+    QPainter painter(&pixmap);
+    pixmap.fill(Qt::black);
+
+    ColorPalette palette = m_colorPaletteSettings.getHotcueColorPalette(paletteName);
+    if (paletteName == palette.getName()) {
+        for (int i = 0; i < palette.size() && i < 4; ++i) {
+            painter.setPen(mixxx::RgbColor::toQColor(palette.at(i)));
+            painter.setBrush(mixxx::RgbColor::toQColor(palette.at(i)));
```

Comment:
Can you use `palette.colorForHotcueIndex()` instead? Right now, the icon may show colors that are not actually assigned to hotcues.